data/transactions/logicsig.go

```diff
@@ -56,6 +56,7 @@ func (lsig *LogicSig) Len() int {
 
 // Equal returns true if both LogicSig are equivalent
 func (lsig *LogicSig) Equal(b *LogicSig) bool {
+	// TODO is this sufficient?
 	if len(lsig.Logic) == 0 && len(b.Logic) == 0 {
 		return true
```

Comment:
Two LogicSigs could both have 0 length Logic but have different Sig, Msig, or Args. Not sure if this causes problems or not.

Reply:
we should save ourselves a future headache and check that all args are exactly the same (including nilness)